tslearn/shapelets.py

```diff
@@ -314,32 +338,59 @@ def fit(self, X, y):
         y : array-like of shape=(n_ts, )
             Time series labels.
         """
+        X, y = check_X_y(X, y, allow_nd=True)
+        X = to_time_series_dataset(X)
+        X = check_dims(X, X_fit=None)
+
         set_random_seed(seed=self.random_state)
         numpy.random.seed(seed=self.random_state)
+
         n_ts, sz, d = X.shape
-        self.d = d
-        if y.ndim == 1:
-            self.label_binarizer = LabelBinarizer().fit(y)
-            y_ = self.label_binarizer.transform(y)
+        self.X_fit_ = X
+
+        self.model_ = None
```

Comment:
Which of these attributes should be documented? Maybe at least `model_`, `transformer_model_` and `locator_model_` ?